### DISCUSSION TITLE
Dirt bombs can bypass region protection

### DISCUSSION THREAD

**Discussion by TomAltonator** [Pryaxis/TShock#2861] [Q&A]
Is there any way to stop dirt bombs spawning dirt in region protected areas?
Friend of mine found he could use a dirt bomb in a region he didn't have permission for.

**Comment by sgkoishi** [comment]
It should be reverted 🤔
Will have a look later

**Reply by sgkoishi** [reply]
Sadly, the logic for tile/liquid bombs are quite different from regular bombs: regular bombs destroy tiles on the client side, and the client send those changes to the server (so that the server can proceed with each change and revert whatever they don't want).
For dirt bombs, the client sends a `ProjectileDestroy` and the server will have those projectiles' AI executed, and the tile actions are within the AI `Projectile.Kill()`
```
if (Main.netMode != 1)
{
    this.Kill_DirtAndFluidProjectiles_RunDelegateMethodPushUpForHalfBricks(
        base.Center.ToTileCoordinates(),
        4.2f,
        DelegateMethods.SpreadDirt);
}
```

We need a hook here (probably via MonoMod's `On.` hook) since it doesn't fit to existing Bouncer or GetDataHandlers 🤔

**Reply by sgkoishi** [reply]
[My misc plugin](https://github.com/sgkoishi/yaaiomni/) got a temporary workaround for dirt and liquid bombs/rockets.
Key snippets [1](https://github.com/sgkoishi/yaaiomni/blob/d55461f151fc898d03024e716343d68314436da6/src/Soundness/ProjectileKill.cs#L11-L84) [2](https://github.com/sgkoishi/yaaiomni/blob/d55461f151fc898d03024e716343d68314436da6/src/Plugin.cs#L62) if someone don't want the prebuilt version.
